src/decorators.ts

```diff
@@ -126,6 +126,32 @@ export function ResponseSchema(
         },
       }
 
+      const oldSchema =
+        source.responses[statusCode]?.content[contentType].schema
+
+      const makeNewSchema = () => {
+        if (oldSchema.oneOf) {
+          return {
+            oneOf: _.concat([...oldSchema.oneOf], schema),
+          }
+        }
+
+        return { oneOf: [{ ...oldSchema }, schema] }
```

Comment:
Do we still need the shallow copy (`{ ...oldSchema }`) here?

Reply:
nope - chalk that up to my unfamiliarity with ts :)